src/main/java/picard/analysis/AlignmentSummaryMetricsCollector.java

```diff
@@ -230,15 +230,21 @@ private void collectReadData(final SAMRecord record) {
                     metrics.PF_READS++;
                     if (isNoiseRead(record)) metrics.PF_NOISE_READS++;
 
-                    if (record.getReadUnmappedFlag()) {
-                        // If the read is unmapped see if it's adapter sequence
+                    if (record.getReadUnmappedFlag() || record.getMappingQuality() == 0) {
+                        // If the read is unmapped (or mapping quality 0) see if it's adapter sequence
                         final byte[] readBases = record.getReadBases();
-                        if (!(record instanceof BAMRecord)) StringUtil.toUpperCase(readBases);
+                        final boolean revComp = !record.getReadUnmappedFlag() &&
```

Comment:
This duplicates the code in `isAdapter()` above, and has the same side effecting issue. Is there a way to refactor the code and avoid the duplication?

Reply:
👍